drivers/sensor/kx022/kx022_trigger.c

```
@@ -0,0 +1,377 @@
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	__ASSERT_NO_MSG(trig->type == SENSOR_TRIG_DELTA);
+
+	if (gpio_pin_interrupt_configure(data->gpio, cfg->irq_pin, GPIO_INT_DISABLE)) {
+		LOG_ERR("Failed INT_DISABLE");
+		return -EINVAL;
+	}
+
+	if (handler == NULL) {
+		LOG_WRN("kx022: no handler");
+	}
+
+	/* re-trigger lost interrupt */
+	if (data->hw_tf->read_data(dev, KX022_REG_XOUT_L, buf, sizeof(buf)) < 0) {
+		LOG_ERR("status reading error");
+		return -EIO;
+	}
+
+	if (gpio_pin_interrupt_configure(data->gpio, cfg->irq_pin, GPIO_INT_EDGE_TO_ACTIVE)) {
+		LOG_ERR("Failed INT_EDGE_TO_ACTIVE");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+int kx022_restore_default_motion_setup(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (kx022_mode(dev, KX022_STANDY_MODE) < 0) {
+		return -EIO;
+	}
+
+	/* reset motion detect function*/
+	if (data->hw_tf->update_reg(dev, KX022_REG_CNTL1, KX022_MASK_CNTL1_WUFE,
+				    KX022_CNTL1_WUFE_RESET) < 0) {
+		LOG_ERR("Failed to disable motion detect");
+		return -EIO;
+	}
+
+	/* reset motion detect int 1 report */
+	if (data->hw_tf->update_reg(dev, KX022_REG_INC4, KX022_MASK_INC4_WUFI1,
+				    KX022_INC4_WUFI1_RESET) < 0) {
+		LOG_DBG("Failed to set KX022 int1 report");
+		return -EIO;
+	}
+
+	kx022_mode(dev, KX022_OPERATING_MODE);
+
+	return 0;
+}
+
+int kx022_restore_default_tilt_setup(const struct device *dev)
+{
+	struct kx022_data *data = dev->data;
+
+	if (kx022_mode(dev, KX022_STANDY_MODE) < 0) {
+		return -EIO;
+	}
+
+	/* reset the tilt function */
+	if (data->hw_tf->update_reg(dev, KX022_REG_CNTL1, KX022_MASK_CNTL1_TPE,
+				    KX022_CNTL1_TPE_RESET) < 0) {
+		LOG_ERR("Failed to disable tilt");
+		return -EIO;
+	}
+
+	/* reset tilt int 1 report */
+	if (data->hw_tf->update_reg(dev, KX022_REG_INC4, KX022_MASK_INC4_TPI1,
+				    KX022_INC4_TPI1_RESET) < 0) {
+		LOG_DBG("Failed to set KX022 tilt int1 report");
+		return -EIO;
+	}
```

in general: propagate error codes if error is not handled.

```
int ret;

ret = f();
if (ret < 0) {
    return ret;
}
```